tests/cnf/core/network/metallb/tests/frrk8-tests.go

```diff
@@ -0,0 +1,760 @@
+					// Get the routes
+					routes, err := frr.VerifyBGPReceivedRoutesOnFrrNodes(frrk8sPods)
+					Expect(err).ToNot(HaveOccurred(), "Failed to verify BGP routes")
+
+					// Return the routes to be checked
+					return routes
+				}, 60*time.Second, 5*time.Second).Should(SatisfyAll(
+					ContainSubstring(externalAdvertisedIPv4Routes[0]), // First IP address
+					ContainSubstring(externalAdvertisedIPv4Routes[1]), // Second IP address
+				), "Fail to find all expected received routes")
+			})
+
+		It("Validate a FRR node receives and sends IPv4 and IPv6 routes from an EBGP multihop FRR instance",
+			reportxml.ID("47279"), func() {
+
+				By("Collecting information before test")
+				frrk8sPods, err := pod.List(APIClient, NetConfig.MlbOperatorNamespace, metav1.ListOptions{
+					LabelSelector: "app=frr-k8s",
+				})
+				Expect(err).ToNot(HaveOccurred(), "Failed to list speaker pods")
+				By("Setting test iteration parameters")
+				_, _, _, nodeAddrList, addressPool, _, err :=
+					metallbenv.DefineIterationParams(
+						ipv4metalLbIPList, ipv6metalLbIPList, ipv4NodeAddrList, ipv6NodeAddrList, "IPv4")
+				Expect(err).ToNot(HaveOccurred(), "Fail to set iteration parameters")
+
+				By("Creating MetalLb Master pod configMap")
+				frrPod := createMasterFrrPod(64501, nodeAddrList, hubIPaddresses, externalAdvertisedIPv4Routes,
+					externalAdvertisedIPv6Routes, true)
+
+				By("Creating BGP Peers")
+				createBGPPeerAndVerifyIfItsReady("172.16.0.1", "", tsparams.RemoteBGPASN,
+					true, frrk8sPods)
+
+				By("Checking that BGP session is established and up")
+				verifyMetalLbBGPSessionsAreUPOnFrrPod(frrPod, removePrefixFromIPList(ipv4NodeAddrList))
+
+				By("Validating BGP route prefix")
+				validatePrefix(frrPod, "IPv4", removePrefixFromIPList(nodeAddrList), addressPool, 32)
+
+				By("Create a frrconfiguration allow all for EBGP multihop")
+				err = createFrrConfiguration("frrconfig-allow-all", "172.16.0.1", 64501, nil, true)
+				Expect(err).ToNot(HaveOccurred(), "Fail to create an allow-all FRRconfiguration")
+
+				By("Verify that the node FRR pods advertises two routes")
+				err = verifyExternalAdvertisedRoutes(frrPod, ipv4NodeAddrList, externalAdvertisedIPv4Routes)
+				Expect(err).ToNot(HaveOccurred(), "Fail to find advertised routes")
+
+				By("Validate BGP received routes")
+				Eventually(func() string {
+					// Get the routes
+					routes, err := frr.VerifyBGPReceivedRoutesOnFrrNodes(frrk8sPods)
+					Expect(err).ToNot(HaveOccurred(), "Failed to verify BGP routes")
+
+					// Return the routes to be checked
+					return routes
+				}, 60*time.Second, 5*time.Second).Should(SatisfyAll(
+					ContainSubstring(externalAdvertisedIPv4Routes[0]), // First IP address
+					ContainSubstring(externalAdvertisedIPv4Routes[1]), // Second IP address
+				), "Fail to find all expected received routes")
+			})
+	})
+
+})
+
+func createNewMetalLbDaemonSetAndWaitUntilItsRunningWithAlwaysBlock(timeout time.Duration,
```

Comment:
Valdate all errors inside the local func via Expect()

Reply:
Changed